### ISSUE TITLE
Form is submitted when required checkboxes are not checked

### ISSUE THREAD

**Issue by indrimuska** [bvaughn/angular-form-for#147]
Hi everybody!
I found this bug while I was working with checkboxes, I don't know if the same problem occurs with radio inputs.

Here is a [plunker](http://plnkr.co/edit/YmPyWsskmv0ioGLSpfd9?p=preview) showing the problem, just perform the following steps:
* try to submit
* check the input
* uncheck the input
* try again to submit the form

You will notice than from now on, the form is always submitted even if the input is not checked. As you will see, also `field-error`s are never shown.

**Comment by bvaughn** [comment]
Thanks for reporting this. As crazy as it seems...this looks like a legit oversight. :) I'll try to patch it up soon.

**Comment by bvaughn** [comment]
In fact, now that I think back, that's how it's implemented on the demo page also:
http://bvaughn.github.io/angular-form-for/#/demo/form-builder-markup

What are your thoughts on the work around?

**Comment by indrimuska** [comment]
Yes, the workaround works perfectly, but still remain a workaround.

I was looking at the normal behavior of a checkbox in Angular.js: https://docs.angularjs.org/examples/example-example34/index.html.

As I can see, Angular sets the model object value to `undefined` every time the checkbox is unchecked. The same should be for the formFor plugin.

I'm debuggin the plunker and i'm trying to understand how (and why) formFor sets the value of the checkbox to true/false instead of true/undefined.

**Comment by indrimuska** [comment]
Ok, this is a normal behavior of Angular checkboxes when there's not a `required` attribute.

That's what I noticed:
* if the checkbox has the `required` attribute, then Angular set the unchecked state to `undefined`
* otherwise Angular set the unchecked state to `false`

Here is a [plunker](http://plnkr.co/edit/PpGVcWgRbEIa8huHXh1o) showing this behavior.

**Comment by indrimuska** [comment]
I think there are many options to solve this:

* require `novalidate` attribute to the form and `required` attribute to the checkbox ([plunker](http://plnkr.co/edit/CFRlVUiUGdeaMWxCZNp8?p=preview))
* working on the plugin to wrap this behavior, adding a watcher [here](https://github.com/bvaughn/angular-form-for/blob/master/dist/form-for.js#L410)
```javascript
$scope.$watch('model.bindable', function (value) {
   $scope.model.bindable = value || undefined; // or null
});
```
* something similar to point 2, but generalized to all inputs (maybe [here](https://github.com/bvaughn/angular-form-for/blob/master/dist/form-for.js#L1045)?). In this case we must check the input type).

I personally prefer the first one, what do you think about?

**Comment by bvaughn** [comment]
I merged in your fix! Although I didn't merge your branch, because you didn't build from source (and so the minified and source-map versions of the dist hadn't been updated). Thank you for the suggestion and the fix though! :)

**Comment by bvaughn** [comment]
The fix for this issue has been released in version [4.1.1](https://github.com/bvaughn/angular-form-for/releases/tag/4.1.1).

**Comment by indrimuska** [comment]
Thank you for supporting me! :+1:
